### DISCUSSION TITLE
System Tests

### DISCUSSION THREAD

**Discussion by partydrone** [nickjj/docker-rails-example#6] [Q&A]
@nickjj, It's great to see you opened up this project for discussion. Something I've been really curious about is:

1. Do you run system tests when coding a project?
2. Do you use this Dockerized project template when you do run system tests?
3. What does that look like/how does that work?

Is there anything you can share from actual projects that shows how system tests are executed? Do you have a blog post about this? Would love to hear your thoughts on the matter.

**Comment by nickjj** [comment]
Hi,

Do you mean running `rails test`? I do typically run it while developing a project with `run rails test`. Not necessarily in TDD fashion but I do write tests.

I do use this Dockerized set up when running tests. You can run your tests with `run rails test`. Basically the same as without Docker. It would be expected that you run a `docker-compose up` beforehand and then the test command ends up getting run inside of the webpacker container (mainly because access to node is required to generate test packs).

The readme has this documented at: https://github.com/nickjj/docker-rails-example#running-the-test-suite

**Reply by partydrone** [reply]
> [System tests](https://guides.rubyonrails.org/testing.html#system-testing) allow you to test user interactions with your application, running tests in either a real or a headless browser. System tests use Capybara under the hood.

System tests are not run when using `rails test` by default. You must use `rails test:system` or `rails test:all` to run them.

**Reply by nickjj** [reply]
Oh. Normally I don't run them but with Hotwire Turbo being a thing along with integrating in Stimulus, it seems like a reasonable idea to start using them more regularly.
